### PR TITLE
EWPP-119: Redirect outbound processor.

### PR DESCRIPTION
**Pull request by sergepavle**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: modules/oe_content_redirect_link_field/tests/src/Kernel/PathProcessorRedirectLinkTest.php [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    $node = $this->container->get('entity_type.manager')->getStorage('node')->load($node->id());
+    $translated_node = $node->getTranslation('fr');
+    $this->assertEquals('http://example.com/fr', $translated_node->get('oe_redirect_link')->uri);
+    $this->assertTrue($node->get('oe_redirect_link')->isEmpty());
+    $url = Url::fromRoute('entity.node.canonical', ['node' => $node->id()], ['language' => $language])->toString();
+    $this->assertEquals($url, $url);
+    $this->assertEquals($url, $translated_node->toUrl()->toString());
+  }
+
+}
```
Comment (brummbar) [inline modules/oe_content_redirect_link_field/tests/src/Kernel/PathProcessorRedirectLinkTest.php:214]: Cache propagation tests are missing.

**Review by brummbar — CHANGES_REQUESTED** [review]

File: modules/oe_content_redirect_link_field/src/PathProcessor/PathProcessorRedirectLink.php [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    $options['external'] = FALSE;
+    $options['prefix'] = '';
+    try {
+      return Url::fromUri($redirect_path, $options)->toString();
+    }
+    catch (\Exception $exception) {
+      return $path;
+    }
```
[the remark this answers is not in the document]
Comment (brummbar) [reply modules/oe_content_redirect_link_field/src/PathProcessor/PathProcessorRedirectLink.php]: Old comment

File: modules/oe_content_redirect_link_field/tests/src/Kernel/PathProcessorRedirectLinkTest.php [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    $node->save();
+    $url = Url::fromRoute('entity.node.canonical', ['node' => $node->id()])->toString();
+    $this->assertEquals('http://example2.com', $url);
+    $this->assertEquals('http://example2.com', $node->toUrl()->toString());
+
+    Role::load($user->getRoles()[1])->grantPermission('bypass redirect link outbound rewriting');
+    $url = Url::fromRoute('entity.node.canonical', ['node' => $node->id()])->toString();
+    // Users with the permission to bypass the rewriting should not see a
+    // different URL.
+    $this->assertEquals($url, $url);
```
[the remark this answers is not in the document]
Comment (upchuk) [reply modules/oe_content_redirect_link_field/tests/src/Kernel/PathProcessorRedirectLinkTest.php:157]: lol. Why not? :)

File: modules/oe_content_redirect_link_field/tests/src/Kernel/PathProcessorRedirectLinkTest.php [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+    Role::load($user->getRoles()[1])->revokePermission('bypass redirect link outbound rewriting');
+    $url = Url::fromRoute('entity.node.canonical', ['node' => $node->id()])->toString();
+    $this->assertEquals('http://example2.com', $url);
+
+    // If there is no redirect link, there should be no bypassing.
+    $node->set('oe_redirect_link', '');
+    $node->save();
+    $url = Url::fromRoute('entity.node.canonical', ['node' => $node->id()])->toString();
+    $this->assertEquals($url, $url);
```
[the remark this answers is not in the document]
Comment (upchuk) [reply modules/oe_content_redirect_link_field/tests/src/Kernel/PathProcessorRedirectLinkTest.php:167]: :))

File: modules/oe_content_redirect_link_field/tests/src/Kernel/PathProcessorRedirectLinkTest.php (hunk 3 above)
[the remark this answers is not in the document]
Comment (sergepavle) [reply modules/oe_content_redirect_link_field/tests/src/Kernel/PathProcessorRedirectLinkTest.php:157]: oh no... it wasn't with evil intention or try to cheat.. just happened... sorry

**Review by brummbar — APPROVED** [review]

**Review by upchuk — APPROVED** [review]

**Pull request merged**